### PR TITLE
Add `rails g ember-cli:heroku`

### PR DESCRIPTION
**Pull request by seanpdoyle**
To configure your Ember CLI Rails app to be ready to deploy on Heroku:

1. Run `rails g ember-cli:heroku` generator
1. [Add the NodeJS buildpack][buildpack] and configure NPM to include the
   `bower` dependency's executable file.

```sh
heroku buildpacks:add --index 1 https://github.com/heroku/heroku-buildpack-nodejs
heroku config:set NPM_CONFIG_PRODUCTION=false
```

[buildpack]: https://devcenter.heroku.com/articles/using-multiple-buildpacks-for-an-app#adding-a-buildpack

### CODE REVIEW

File: lib/generators/ember-cli/heroku/heroku_generator.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+module EmberCLI
+  class HerokuGenerator < Rails::Generators::Base
+    source_root File.expand_path("../templates", __FILE__)
+
+    namespace "ember-cli:heroku"
+
+    gem "rails_12factor", group: [:staging, :production]
```
Comment (seanpdoyle) [inline lib/generators/ember-cli/heroku/heroku_generator.rb]: This is causing a strange error:

```
/Users/seanpdoyle/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/bundler-1.10.6/lib/bundler/rubygems_integration.rb:292:in `bloc
k in replace_gem': rails_12factor is not part of the bundle. Add it to Gemfile. (Gem::LoadError)
```

[gem](http://api.rubyonrails.org/classes/Rails/Generators/Actions.html#method-i-gem) is a generator method, but this looks like it's using the version that executes in the `Gemfile`

File: README.md [hunk 2]
```diff
@@ -301,50 +301,17 @@ directories %w[app config lib spec your-appname/app]
...
-In order to deploy Ember CLI Rails app to Heroku:
+To configure your Ember CLI Rails app to be ready to deploy on Heroku:
 
-First, enable Heroku Multi Buildpack by running the following command:
+1. Run `rails g ember-cli:heroku` generator
+1. Enable Heroku Multi Buildpack by running the following commands:
 
 ```sh
+heroku config:set NPM_CONFIG_PRODUCTION=false
 heroku buildpacks:set https://github.com/heroku/heroku-buildpack-multi
```
Comment (geoffharcourt) [inline README.md]: From the [recent docs](https://devcenter.heroku.com/articles/using-multiple-buildpacks-for-an-app#adding-a-buildpack) it looks like you can just use `heroku buildpacks:add` and don't need to declare the Multi buildpack anymore.

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,8 @@ master
 ------
 
 * Drop support for Ruby `< 2.1.0` and Rails `4.0.0, < 3.2.0`
+* Introduce `rails g ember-cli:heroku` generator to setup a project for
```
Comment (geoffharcourt) [inline CHANGELOG.md]: I made this comment and then Github had some saving problems.

How about "to set up a project for Heroku deployment"?  ("setup" is a noun, "set up" is a verb)

**Comment by tute** [comment]
Looks great to me! I'm excited about this gem. :)

**Pull request merged**